utils/config_generator.py

```diff
@@ -0,0 +1,383 @@
+):
+    print("Fetching opus")
+    opus_datasets = fetch_opus(source, target)
+    total_sentences = 0
+    skipped_datasets = []
+    visited_corpora = set()
+
+    for dataset in opus_datasets:
+        # Some datasets are ignored or too small to be included.
+        if dataset.corpus in skip_datasets:
+            skipped_datasets.append(f"{dataset.corpus_key()} - ignored datasets")
+            continue
+        if (dataset.alignment_pairs or 0) < minimum_dataset_sentences:
+            skipped_datasets.append(f"{dataset.corpus_key()} - not enough data")
+            continue
+
+        visited_corpora.add(normalize_corpus_name(dataset.corpus))
+        total_sentences += dataset.alignment_pairs or 0
+        corpus_key = dataset.corpus_key()
+        train_datasets.append(corpus_key)
+        train_datasets.yaml_add_eol_comment(
+            f"{dataset.alignment_pairs:,} sentences".rjust(70 - len(corpus_key), " "),
+            len(train_datasets) - 1,
+        )
+
+    print("Fetching mtdata")
+    entries = fetch_mtdata(source, target)
+
+    for corpus_key, entry in entries.items():
+        corpus_name = normalize_corpus_name(entry.did.name)
+        group_corpus_name = normalize_corpus_name(entry.did.group + entry.did.name)
+        if not (corpus_name in visited_corpora or group_corpus_name in visited_corpora):
+            train_datasets.append(corpus_key)
+            byte_size, display_size = get_remote_file_size(entry.url)
+            if byte_size:
+                # Don't add the sentences to the total, as these will be commented out by default.
+                sentences = estimate_sentence_size(byte_size)
+                train_datasets.yaml_add_eol_comment(
+                    f"~{sentences:,} sentences ".rjust(70 - len(corpus_key), " ")
+                    + f"({display_size})",
+                    len(train_datasets) - 1,
+                )
+
+    train_comment = "\n".join(
+        [
+            "The training data contains:",
+            f"  {total_sentences:,} sentences",
+            "",
+            "Skipped datasets:",
+            *[f" - {d}" for d in skipped_datasets],
+        ]
+    )
+
+    comment_section["  train:"] = train_comment
+
+
+def normalize_corpus_name(corpus_name: str):
+    """Normalize the corpus name so that it's easy to deduplicate between opus and mtdata."""
+
+    # Remove the language tags at the end.
+    # mtdata_ELRC-vnk.fi-1-eng-fin
+    #                     ^^^^^^^^
+    corpus_name = re.sub(r"-\w{3}-\w{3}$", "", corpus_name)
+
+    corpus_name = corpus_name.lower()
+
+    # Remove numbers anything that is not a letter. This is a little aggressive, but should help
+    # deduplicate more datasets. For example:
+    #   opus: 725-Hallituskausi_2011_2
+    #   mtdata: hallituskausi_2011_2015-1-eng-fin
+    corpus_name = re.sub(r"[^a-z]", "", corpus_name.lower())
+
+    # Ignore train/test/dev splits.
+    #   opus: NeuLab-TedTalks/v1
+    #   mtdata: Neulab-tedtalks_train-1-eng-fin
```

Comment:
Not sure if it's relevant here, but if such a dataset with 3 parts is present we usually use it in `train/devtest/test`